app/src/androidTest/java/com/ea/themovie/repository/MockMovieRepositoryImp.java

```diff
@@ -0,0 +1,60 @@
+package com.ea.themovie.repository;
+
+import android.util.Log;
+
+import com.ea.themovie.api.Error;
+import com.ea.themovie.api.MovieApi;
+import com.ea.themovie.entity.MovieList;
+import com.ea.themovie.entity.MovieVideo;
+import com.ea.themovie.util.Constants;
+import com.ea.themovie.util.TestData;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class MockMovieRepositoryImp extends MovieRepositoryImp{
+    @Override
+    public void getPopularMovies(int page, final MovieRepositoryCallback<MovieList> callback) {
```

Comment:
If callback is null, we can just throw an exception instead of check it every time. Btw, as this is a mock class and data is retrieve via callback, we can simply let is throw a nullpointer exception here.

A better implementation of this is using RxJava.
```
Observable.create(...).doOnSubscribe({
    // load the list of movies
    // emit it to subscribers
})
```

Reply:
Yes, i will apply RxJava in later task